src/model/Request.php

```diff
@@ -5,18 +5,45 @@
 // hnusna skareda staticka vec! fuj!
 class Request
 {
-    public static function getIp()
+    public static function getIp(): string
     {
-        if (!empty($_SERVER['HTTP_CLIENT_IP'])) {
-            $ip = $_SERVER['HTTP_CLIENT_IP'];
-        } elseif (!empty($_SERVER['HTTP_X_FORWARDED_FOR']) && $_SERVER['HTTP_X_FORWARDED_FOR'] != '127.0.0.1') {
-            $ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
-        } elseif (!empty($_SERVER['REMOTE_ADDR'])) {
-            $ip = $_SERVER['REMOTE_ADDR'];
-        } else {
-            $ip = 'cli';
+        $serverKeysToCheck = [
+            'HTTP_R_FORWARDED_FOR', // special header for custom IP if needed
+            'HTTP_X_FORWARDED_FOR',
+            'HTTP_X_FORWARDED',
+            'HTTP_FORWARDED',
+            'REMOTE_ADDR',
+            'HTTP_CLIENT_IP',
+        ];
+
+        foreach ($serverKeysToCheck as $serverKey) {
+            if (!self::validateAddress($serverKey)) {
+                continue;
+            }
+            return self::extractFirstAddress($serverKey);
```

Comment:
This extraction already happened in `validate`. Wouldn't it be sufficient to return the IP address directly from there? (string/null instead of returning true/false)